packages/astro/src/transitions/router.ts

```diff
@@ -213,7 +213,9 @@ const moveToLocation = (to: URL, from: URL, options: Options, historyState?: Sta
 			// ... what comes next is a intra-page navigation
 			// that won't reload the page but instead scroll to the fragment
 			history.scrollRestoration = 'auto';
-			location.href = to.href;
+			const savedState = history.state;
```

Comment:
I think `state` is an object right? So shouldn't we make a copy instead of a reference using `{ ...history.state }`?

Reply:
Yes, it is an object. It looks like Firefox just clears the reference and leaves the object alone. I think it's safe to keep the reference alive and restore it after the assignment. Why would Firefox bother changing the attributes of a custom object that it doesn't even know what attributes it has?